api/coreservice.go

```diff
@@ -108,6 +108,8 @@ type (
 		ActionsByAddress(addr address.Address, start uint64, count uint64) ([]*iotexapi.ActionInfo, error)
 		// ActionByActionHash returns action by action hash
 		ActionByActionHash(h hash.Hash256) (action.SealedEnvelope, hash.Hash256, uint64, uint32, error)
+		// PendingActionByActionHash returns action by action hash
+		PendingActionByActionHash(h hash.Hash256) (action.SealedEnvelope, error)
```

Comment:
utilize `Action(actionHash string, checkPending bool) (*iotexapi.ActionInfo, error)`?

Reply:
A great suggestion. However, I noticed that the coreservice provides two interfaces with duplicate functionality:
```
Action(actionHash string, checkPending bool) (*iotexapi.ActionInfo, error)
ActionByActionHash(h hash.Hash256) (action.SealedEnvelope, hash.Hash256, uint64, uint32, error)
```

The returned action structures are different. The first is currently used for gRPC, while the second is used for web3. Therefore, I added an interface specifically for web3 to query pending actions, to accommodate web3's processing logic:
```
PendingActionByActionHash(h hash.Hash256) (action.SealedEnvelope, error)
```

Reply:
However, it seems that coreservice should provide a unified interface for both gRPC and web3. We can merge into a unified function in a separate PR.

Reply:
yes, let's do that in another PR

Reply:
add TODO for the improvement